tests/Storage/Device/LocalTest.php

```diff
@@ -221,4 +221,50 @@ public function testPartitionTotalSpace()
     {
         $this->assertGreaterThan(0, $this->object->getPartitionTotalSpace());
     }
+
+    /**
+     * @depends testPartUpload
+     */
+    public function testTransferLarge($path) {
+        
+        // chunked file
+        $this->object->setTransferChunkSize(10000000); //10 mb
+        
+        $key = $_SERVER['S3_ACCESS_KEY'] ?? '';
+        $secret = $_SERVER['S3_SECRET'] ?? '';
+        $bucket = 'appwrite-storage';
+
+        $device = new S3('/root', $key, $secret, $bucket, S3::AP_SOUTH_1, S3::ACL_PRIVATE);
+        $destination = $device->getPath('largefile.mp4');
+
+        $this->assertTrue($this->object->transfer($path, $destination, $device ));
+        $this->assertTrue($device->exists($destination));
+        $this->assertEquals($device->getFileMimeType($destination), 'video/mp4');
+
+        $device->delete($destination);
+        $this->object->delete($path);
+
+    }
+
+    public function testTransferSmall() {
```

Comment:
We are only testing Local -> S3/DO.
Should we also test DO/S3 -> Local, and maybe S3 <-> DO?

Reply:
yes S3 -> Local tests are being worked on